Draft: Networking

Martin554 opened the pull request.

Martin554 committed:
disable spawning via spawner

Martin554 committed:
merge from david

Martin554 committed:
correct parenting and postioning of network components

Martin554 committed:
merge from david

Martin554 committed:
graph fix

Martin554 committed:
set names

Martin554 committed:
fixes + nodeName at client side

Martin554 committed:
ClassDiagramBuilder is strategy

Martin554 committed:
ClassDiagramBuilderFactory

Martin554 committed:
set added objects network name

Martin554 committed:
create graph from client

Martin554 committed:
Assign graph into diagram pool at client side

Martin554 committed:
merge from david

Martin554 committed:
IVisualEditor, visual editor non-static

Martin554 committed:
Fix of graph and nodes creation when networking is disabled

Martin554 committed:
network edges

Martin554 committed:
network graph script with update line points RPC

Martin554 committed:
network line caps

Martin554 committed:
attributes, methods sent from server to clients

Martin554 committed:
Update node name

Martin554 committed:
merge from develop

Martin554 committed:
fix after merge

Martin554 committed:
Delete button as prefab of UEdge

Martin554 committed:
Fix of updating of non-spawned class

Martin554 committed:
update method, attributes server->client

Martin554 committed:
delete method, attribute server->client

Martin554 committed:
line endings, MainEditor client and server files

Martin554 committed:
MainEditorFactory, main editor and diagram builder have visual editor

Martin554 committed:
remove node client->server

Martin554 committed:
Delete relation client->server

Martin554 committed:
merge of add method in MainEditor

Martin554 committed:
AddMethod client->Server

Martin554 committed:
Add attribute client->server

Martin554 committed:
delete method, attribute client->server

Martin554 committed:
update name, method, attribute client->server

Martin554 committed:
remove of unused mainEditorServer

Martin554 committed:
One createNode mothod

Martin554 committed:
createClass client->server, NextClassId from ClassDiagram

Martin554 committed:
Client: added class to has active buttons in editing mode

Martin554 committed:
Client->server sync of class positions

Martin554 committed:
merge from develop

Martin554 committed:
fix after merge

Martin554 committed:
enable edit button

Martin554 committed:
merge from develop

Martin554 committed:
affter merge

dwwop commented on UnityProjectDP/Assets/Scripts/Visualization/UI/UIEditorManager.cs:
```diff
@@ -13,40 +15,48 @@ public class UIEditorManager : Singleton<UIEditorManager>
         private GameObject _fromClass;
         private string _relType;
+        private IClassDiagramBuilder _classDiagramBuilder;
+        public MainEditor mainEditor;
```
Nemyslim si, ze by instancie mainEditor a classDiagramBuilder mali byt v tejto class. Jej pointa je obsluhovat UI, tieto dve su viac vseobecne, lepsie by mi prislo z nich spravit singletony a pri prvom get ich inicializovat

dwwop commented on UnityProjectDP/Assets/Scripts/Visualization/ClassDiagram/IClassDiagramBuilder.cs:
```diff
@@ -0,0 +1,20 @@
+{
+    public class IClassDiagramBuilder
+    {
+        public IVisualEditor visualEditor;
```
Lepsie by bolo spravit visual editor v main editore public, a mat ho ulozeneho len na jednom mieste

dwwop commented on UnityProjectDP/Assets/Scripts/Visualization/UI/UIEditorManager.cs:
```diff
@@ -13,40 +15,48 @@ public class UIEditorManager : Singleton<UIEditorManager>
         {
             DontDestroyOnLoad(gameObject);
+            _classDiagramBuilder = ClassDiagramBuilderFactory.Create();
+            mainEditor = MainEditorFactory.Create(_classDiagramBuilder.visualEditor);
```
Ako pisane v IClassDiagramBuilder, spravit visual editor v main editor public a nechat classDiagramEditor k nemu pristupovat tak

dwwop committed:
Merge branch 'networking' into david/editing-class-and-object-layers

dwwop committed:
Various UI fixes

- Remove duplicate FileBrowser
- Refactor Tool Manager
- PopUp setButtonsActive fix
- Class prefab change Name to unconstrained size

dwwop committed:
Adjust Namespaces to File Structure

dwwop committed:
Deep edit, user input validation and bug fixes

Deep edit of Class Names
Zmena generovania idciek
Osetrenie uzivatelskeho vstupu
Oprava error labels

dwwop committed:
Merge branch 'networking' into david/editing-class-and-object-layers

Martin554 committed:
fix class networked postion from editor

Martin554 committed:
fix networked scale after create from editor

Martin554 committed:
remove of source

Martin554 committed:
fix of relation redraw after adding - quick fix was replaced by graph reroute in diagram loading - call of reroute in MainEditor.CreateRelation

dwwop committed:
Bugfixes and improvements

Error popUp when creating relation
Improved error labels with more errors
Tool Manager Reset Button
Enter on Input field to add component from popup
Ctrl + mousewheel to zoom

dwwop committed:
Changes based on comments from pull req

Martin554 committed:
Merge pull request #12 from dwwop/david/editing-class-and-object-layers

David/editing class and object layers

Martin554 committed:
merge from remote

dwwop committed:
Fix exception when creating animation

Martin554 committed:
fixes after merge

Martin554 committed:
Merge branch 'david/editing-class-and-object-layers' into networking

dwwop committed:
Parsers overhaul

Martin554 committed:
network class created

dwwop committed:
Merge branch 'networking' into david/editing-class-and-object-layers

dwwop committed:
Update position when moving classes fix

Martin554 committed:
Todo

Martin554 committed:
Merge branch 'david/editing-class-and-object-layers' into networking

dwwop committed:
Minor bug fixes

End Cap not showing
Animation at wrong place showing
Visual editor instead of MainEditor call

Martin554 committed:
Merge branch 'david/editing-class-and-object-layers' into networking

Martin554 committed:
change to unity transport, ip and port

Martin554 committed:
show port and ip on screen

Martin554 committed:
renameClassPopup

Martin554 committed:
split add-edit popUp into separated popups

Martin554 committed:
AddAttribute, renameAttribute, rename class pop ups

Martin554 committed:
Paramters and methods pop-ups

Martin554 committed:
method and parameters popups

Martin554 committed:
fix of switching between edit/add method dialog from edit/add paramters dialog

Martin554 committed:
Client side node selection in create diagram

Martin554 committed:
enable edit diagram after load at srever

Martin554 committed:
fix of redrwing generalization caps
